packages/core/src/lib/sign-in-experience.test.ts

```diff
@@ -59,6 +71,58 @@ describe('validate branding', () => {
   });
 });
 
+describe('validate language info', () => {
+  it('should call findAllCustomLanguageKeys', async () => {
+    await validateLanguageInfo({
+      autoDetect: true,
+      fallbackLanguage: 'zh-CN',
+      fixedLanguage: 'en',
+    });
+    expect(findAllCustomLanguageKeys).toBeCalledTimes(1);
+  });
+
+  it('should pass when the language is built-in supported', async () => {
+    const builtInSupportedLanguage = 'tr-TR';
+    await expect(
+      validateLanguageInfo({
+        autoDetect: true,
+        fallbackLanguage: builtInSupportedLanguage,
+        fixedLanguage: 'en',
+      })
+    ).resolves.not.toThrow();
+    expect(findAllCustomLanguageKeys).toBeCalledTimes(1);
+  });
+
+  it('should pass when the language is custom supported', async () => {
+    const customSupportedLanguage = 'zh-HK';
+    findAllCustomLanguageKeys.mockResolvedValueOnce([customSupportedLanguage]);
+    await expect(
+      validateLanguageInfo({
+        autoDetect: true,
+        fallbackLanguage: customSupportedLanguage,
+        fixedLanguage: 'en',
+      })
+    ).resolves.not.toThrow();
+    expect(findAllCustomLanguageKeys).toBeCalledTimes(1);
+  });
+
+  it('unsupported fallback language should fail', async () => {
+    const unsupportedLanguage = 'zh-MO';
```

Comment:
ditto.

Reply:
Refactored. Assert that zh-MO is unsupported.

```ts
    const unsupportedLanguage = 'zh-MO';
    expect(unsupportedLanguage in builtInLanguages).toBeFalsy();
    expect(allCustomLanguageKeys.includes(unsupportedLanguage)).toBeFalsy();
```